lib/galaxy/datatypes/triples.py

```diff
@@ -156,3 +160,28 @@ def set_peek( self, dataset, is_multi_byte=False ):
         else:
             dataset.peek = 'file does not exist'
             dataset.blurb = 'file purged from disk'
+
+
+class HDT( binary.Binary, Triples ):
+    """
+    The HDT triple data format
+    """
+    edam_format = "format_2376"
+    file_ext = "hdt"
+
+    def sniff( self, filename ):
+        with open(filename, "rb") as f:
+            if f.readline( 1024 ).startswith("$HDT"):
```

Comment:
`f.read(4) == "$HDT"` instead of `f.readline( 1024 ).startswith("$HDT")` should be enough here?